src/core/contracts/registry.test.ts

```diff
@@ -312,9 +314,190 @@ describe("registry", () => {
+
+      setupSnapshot();
+
+      beforeAll(async () => {
+        const contractOwner = await signer.getAddress();
+        const identityContract = await new Identity__factory(signer).deploy(contractOwner);
+        await identityContract.deployed();
+      });
+
+      it("returns an empty array", async () => {
+        const result = await getRegistrationsByIdentityAddress(identityContract);
+
+        expect(result).toEqual([]);
+      });
+    });
+
+    describe("when identity is associated (belongs) to one registration", () => {
+      let identityContractAddress: EthereumAddress;
+      const handle = "Joe";
+      const newHandle = "JoeNoSe";
+
+      beforeAll(async () => {
+        await snapshotHardhat(provider);
+      });
+
+      afterAll(async () => {
+        await revertHardhat(provider);
+      });
+
+      beforeAll(async () => {
+        const contractOwner = await signer.getAddress();
+        const identityContract = await new Identity__factory(signer).deploy(contractOwner);
+        await identityContract.deployed();
+
+        identityContractAddress = identityContract.address;
+        await register(identityContractAddress, handle);
+      });
+
+      it("returns one registration", async () => {
+        const result = await getRegistrationsByIdentityAddress(identityContractAddress);
+
+        expect(result.length).toEqual(1);
+        expect(result[0]).toEqual(
+          expect.objectContaining({
+            handle: "Joe",
+            contractAddr: identityContractAddress,
+            dsnpUserURI: expect.any(String),
+          })
+        );
+      });
+
+      describe("and registration handle is updated", () => {
+        beforeAll(async () => {
+          await changeHandle(handle, newHandle);
+        });
+
+        it("returns one registration with updated handle", async () => {
+          const result = await getRegistrationsByIdentityAddress(identityContractAddress);
+          expect(result.length).toEqual(1);
+          expect(result[0]).toEqual(
+            expect.objectContaining({
+              handle: "JoeNoSe",
+              contractAddr: identityContractAddress,
+              dsnpUserURI: expect.any(String),
+            })
+          );
+        });
+
+        describe("and a new identity address is set to registration", () => {
+          let newIdentityContract: EthereumAddress;
+
+          beforeAll(async () => {
+            const contractOwner = await signer.getAddress();
+            const identityContract = await new Identity__factory(signer).deploy(contractOwner);
+            await identityContract.deployed();
+            newIdentityContract = identityContract.address;
+            await changeAddress(newHandle, newIdentityContract);
+          });
+
+          it("returns zero registration associated with old identity", async () => {
+            const result = await getRegistrationsByIdentityAddress(identityContractAddress);
+
+            expect(result).not.toEqual(
```

Comment:
Shouldn't you expect an empty array here?

Reply:
added another expection